lib/src/modules/task.cc

```diff
@@ -510,12 +510,19 @@ void Task::processOutputAndUpdateMetadata(ActionResponse& response)
     std::string &output = response.output.std_out;
 
     if (isValidUTF8(output)) {
-        lth_jc::JsonContainer stdout_result;
-        stdout_result.set("output", output);
+        // Return all relevant results: exitcode, stdout, stderr.
+        lth_jc::JsonContainer result;
+        result.set("exitcode", response.output.exitcode);
+        if (!output.empty()) {
+            result.set("stdout", output);
+        }
+        if (!response.output.std_err.empty()) {
+            result.set("stderr", response.output.std_err);
```

Comment:
Should we sanitize this to ensure it is valid UTF-8?

Reply:
Would would also want to do that in the else clause. Maybe, but I'm tempted to say it's unlikely we'll get binary data on stderr.

Reply:
Fair enough.